server/src/main/java/io/crate/geo/GeoJSONUtils.java

```diff
@@ -290,17 +290,15 @@ public static Map<String, Object> sanitizeMap(Map<?, ?> value) {
                 }
             });
             Map<String, Object> transformed = new HashMap<>();
-            if (sameType[0] == true) {
-                String targetType = COMPOSABLE_TYPES.get(typeInCollection[0]);
-                if (targetType != null) {
-                    transformed.put(TYPE_FIELD, targetType);
-                    final List<Object> coords = new ArrayList<>();
-                    forEach(geometries, input -> {
-                        Map<String, Object> validatedShape = (Map<String, Object>) input;
-                        coords.add(validatedShape.get(COORDINATES_FIELD));
-                    });
-                    transformed.put(COORDINATES_FIELD, coords);
-                }
+            String targetType;
+            if (sameType[0] == true && (targetType = COMPOSABLE_TYPES.get(typeInCollection[0])) != null) {
```

Comment:
Minor suggestion to improve readability:
```suggestion
            String targetType = COMPOSABLE_TYPES.get(typeInCollection[0]);
            if (sameType[0] == true && targetType != null) {
```